app/src/main/java/com/example/indoor_volleyball/Activities/MainActivity.java

```diff
@@ -71,8 +73,12 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
         });
         //Default
         binding.bottomNavigation.setSelectedItemId(R.id.events);
+    }
 
-
+    public void goToGymDetails(Gym gym) {
+        Intent i = new Intent(this, GymDetailActivity.class);
+        i.putExtra("gym", Parcels.wrap(gym));
```

Comment:
I don't like that the key for this info ("gym") is hardcoded in two different places.  If you ever changed it, you'd need to remember to change it at both points.  The way I usually handle this is put a staticMethod on the activity.  It would look like this in your activity class

```
private static final String GYM_KEY = "gym";

public static Intent newIntent(Context context, Gym gym) {
    Intent i = new Intent(context, GymDetailActivity.class);
    i.putExtra(GYM_KEY, Parcels.wrap(gym));
}
```

Then in `onCreate` you'd use the constant from above to get the gym from the intent

And this method would look like:

```
public void goToGymDetails(Gym gym) {
    Intent i = GymDetailActivity.newIntent(this, gym)
    startActivity(i);
}
```

Reply:
This way that activity controls how it wants its intents to be built.  And if you want to launch it from multiple places, you don't need to remember how to build its intent each time.